sdk-core/src/main/java/care/data4life/sdk/network/model/EncryptedRecord.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2020 D4L data4life gGmbH / All rights reserved.
+ *
+ * D4L owns all legal rights, title and interest in and to the Software Development Kit ("SDK"),
+ * including any intellectual property rights that subsist in the SDK.
+ *
+ * The SDK and its documentation may be accessed and used for viewing/review purposes only.
+ * Any usage of the SDK for other purposes, including usage for the development of
+ * applications/third-party applications shall require the conclusion of a license agreement
+ * between you and D4L.
+ *
+ * If you are interested in licensing the SDK for your own applications/third-party
+ * applications and/or if you’d like to contribute to the development of the SDK, please
+ * contact D4L by email to help@data4life.care.
+ */
+package care.data4life.sdk.network.model
+
+import care.data4life.sdk.network.model.NetworkModelContract.EncryptedRecord.Companion.DEFAULT_COMMON_KEY_ID
+import com.squareup.moshi.Json
+
+
+data class EncryptedRecord(
+        @field:Json(name = "common_key_id")
+        private var _commonKeyId: String?,
```

Comment:
maybe instead of using `_commonKeyId` what about

```
        @Json(name = "common_key_id")
        override val commonKeyId: String = DEFAULT_COMMON_KEY_ID,
```
and remove 

```
    init {
        _commonKeyId = _commonKeyId ?: DEFAULT_COMMON_KEY_ID
    }

    override val commonKeyId: String
        get() = this._commonKeyId!!
```

Reply:
This is indeed not pretty. But the original intend is a bit different. It seems it was assumed that on unmarshalling the Record it might happen that the commonKey is null. I am not sure if this can be the case. If not we even do not need a default value. However if it is the case, then the input needs to be nullable, I guess, and the default value will have not really an effect.

Reply:
Looking at the phdp-records service api the commonKeyId is mandatory, but when reviewing iOS it optional there too